courses/migrations/0037_make_platform_non_nullable.py

```diff
@@ -0,0 +1,67 @@
+# Generated by Django 3.2.21 on 2023-10-05 23:44
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+DEFAULT_PLATFORM_NAME = "xPRO"
+
+
+def setup_course_platform(apps, schema_editor):
+    """Associate a platform to all the courses without a platform"""
+
+    Platform = apps.get_model("courses", "Platform")
+    Course = apps.get_model("courses", "Course")
+
+    default_platform, _ = Platform.objects.get_or_create(name=DEFAULT_PLATFORM_NAME)
+    courses_without_platform = Course.objects.filter(platform__isnull=True)
+    for course in courses_without_platform:
+        course.platform = default_platform
+
+    Course.objects.bulk_update(courses_without_platform, ["platform"])
+
+
+def setup_program_platform(apps, schema_editor):
+    """Associate a platform to all the programs without a platform"""
+
+    Platform = apps.get_model("courses", "Platform")
+    Program = apps.get_model("courses", "Program")
+
+    default_platform, _ = Platform.objects.get_or_create(name=DEFAULT_PLATFORM_NAME)
+    programs_without_platform = Program.objects.filter(platform__isnull=True)
+    for program in programs_without_platform:
+        program.platform = default_platform
+
+    Program.objects.bulk_update(programs_without_platform, ["platform"])
```

Comment:
You can leave or take this suggestion. We could have created the helper method with an extra argument named `model` and passed the model `Course` | `Program`. `model` would have replaced `Course` | `Program` and that would have reduced duplication.

Reply:
Good idea, I think we should do this. I'll update you once it's done.